l1-contracts/src/core/Rollup.sol

```diff
@@ -973,6 +1023,51 @@ contract Rollup is EIP712("Aztec Rollup", "1"), Leonidas, IRollup, ITestRollup {
     }
   }
 
+  /**
+   * @notice  Validate an L2 block's blob. TODO: edit for multiple blobs per block
+   * Input bytes:
+   * input[:32]     - versioned_hash
+   * input[32:64]   - z
+   * input[64:96]   - y
+   * input[96:144]  - commitment C
+   * input[144:192] - proof (a commitment to the quotient polynomial q(X))
+   *  - This can be relaxed to happen at the time of `submitProof` instead
+   *
+   * @param blobInput - The above bytes to verify a blob
+   */
+  function _validateBlob(bytes calldata blobInput) internal view returns (bytes32 blobHash) {
+    if (!checkBlob) {
+      return bytes32(blobInput[0:32]);
+    }
+    assembly {
+      blobHash := blobhash(0)
+    }
+    require(blobHash == bytes32(blobInput[0:32]), Errors.Rollup__InvalidBlobHash(blobHash));
+
+    // Staticcall the point eval precompile https://eips.ethereum.org/EIPS/eip-4844#point-evaluation-precompile :
+    (bool success,) = address(0x0a).staticcall(blobInput);
+    require(success, Errors.Rollup__InvalidBlobProof(blobHash));
+  }
+
+  /**
+   * @notice  Converts a BLS12 field element from bytes32 to a nr BigNum type
+   * The nr bignum type for BLS12 fields is encoded as 3 nr fields - see blob_public_inputs.ts:
+   * firstLimb = last 15 bytes;
+   * secondLimb = bytes 2 -> 17;
+   * thirdLimb = first 2 bytes;
+   * Used when verifying epoch proofs to gather public inputs.
+   * @param input - The field in bytes32
+   */
+  function bytes32ToBigNum(bytes32 input)
```

Comment:
The linter should be screaming at you down here as function and parameter names are not following the rules.

Reply:
It's not - could you please advise what best to do here?

Reply:
Should be a versioning, but for internal or private functions use `_` as prefix for the function name, and for parameters generally use `_` as prefix.